### PR TITLE
Maggy blog post

### PR DESCRIPTION
**Pull request by maggyb**
Can someone pr please to make sure all the information is correct or if I should add anything. Thanks

### CODE REVIEW

File: _posts/2016-05-24-journal-of-a-junior-bem.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+BEM - which stands for block, element, modifier - is a front-end naming methodology that the guys at Yandex thought up. I had never heard of BEM syntax before I had started working here at Space Between, and when I first started to look into it, overwhelmed was an understatement! So I dedicated my investment time (which is a beautiful thing)  to it on Friday - and now it makes complete sense and I can’t wait to start using it in future projects. I’m writing this blog post to a) give myself a better understanding and b) hopefully help someone else trying to get their head around it!
+
+###So what exactly is BEM?
+
+BEM is a strict way of arranging your CSS into independent and informative modules. It usually follows this pattern:
+
+<pre><code class="scss">
+.block{}
+.block__element{}
+.block--modifier{}
```
Comment (rafriki) [inline _posts/2016-05-24-journal-of-a-junior-bem.md]: .block__element--modifier{}
i think :)

File: _posts/2016-05-24-journal-of-a-junior-bem.md [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+date:           2016-05-25
+author_id:      "maggy_burns"
+description:    "Hello, Maggy here! I’m going to be updating the SB website with journal posts about all the new technologies i’m learning (if I can keep up) and the first post is going to be about BEM syntax. "
+categories:     space between
+banner:         "/assets/images/blog/radcatbem.jpg"
+---
+
+Hello, Maggy here! I’m going to be updating the SB website with journal posts about all the new technologies i’m learning (if I can keep up) and the first post is going to be about BEM syntax. 
+
+BEM - which stands for block, element, modifier - is a front-end naming methodology that the guys at Yandex thought up. I had never heard of BEM syntax before I had started working here at Space Between, and when I first started to look into it, overwhelmed was an understatement! So I dedicated my investment time (which is a beautiful thing)  to it on Friday - and now it makes complete sense and I can’t wait to start using it in future projects. I’m writing this blog post to a) give myself a better understanding and b) hopefully help someone else trying to get their head around it!
```
Comment (rafriki) [inline _posts/2016-05-24-journal-of-a-junior-bem.md]: double space after beautiful thing)

File: _includes/head.html [hunk 3]
```diff
@@ -69,4 +69,8 @@
     <link rel="canonical" href="{{page.url | replace:'index.html','' | prepend: site.baseurl | prepend: site.url}}">
     <link href='https://fonts.googleapis.com/css?family=Merriweather:400,700|Lato:400,700,300' rel='stylesheet' type='text/css'>
     <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/font-awesome/4.4.0/css/font-awesome.min.css">
+    
+    <script src="http://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.4.0/highlight.min.js"></script>
```
Comment (lukefrake) [inline _includes/head.html]: Can we move these scripts into the [footer, please](https://developer.yahoo.com/blogs/ydn/high-performance-sites-rule-6-move-scripts-bottom-7200.html).

File: _posts/2016-05-24-journal-of-a-junior-bem.md [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+BEM is a strict way of arranging your CSS into independent and informative modules. It usually follows this pattern:
+
+<pre><code class="scss">
+.block{}
+.block__element{}
+.block--modifier{}
+
+</code></pre>
+<br>
+A block represent an object on your website, and could be a form, header, or navbar for example.
```
Comment (rafriki) [inline _posts/2016-05-24-journal-of-a-junior-bem.md]: "block represents"

File: _posts/2016-05-24-journal-of-a-junior-bem.md (hunk 2 above)
Comment (lukefrake) [inline _posts/2016-05-24-journal-of-a-junior-bem.md]: Please link investment time to here - http://spacebetween.io/space/between/2016/04/20/Investment-time.html

File: _posts/2016-05-24-journal-of-a-junior-bem.md [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+A block element represents a descendant of your block and is indicated by beginning with a double underscore ‘__ ‘. Elements are what make up your block, and so can be buttons, textboxes or media items.
```
Comment (rafriki) [inline _posts/2016-05-24-journal-of-a-junior-bem.md]: "A block element" -> "An element"

File: _posts/2016-05-24-journal-of-a-junior-bem.md (hunk 1 above)
Comment (lukefrake) [inline _posts/2016-05-24-journal-of-a-junior-bem.md:22]: Both are correct, you might want to add Johns example for completeness

File: _posts/2016-05-24-journal-of-a-junior-bem.md [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+A  modifier is a different version of your element or block and is indicated by double hyphen ‘--’. It could be something like --blue that would give a style like background-color: blue. It’ll indicate that the object would be blue (yes, really!). It can also identify states like booleans, or values like bold, normal or thin for properties like font weight.
+
+The reason for double underscores and hyphens is because if your block or element has a hyphen in its name then you don’t want to confuse the name with a modifier. Like if you had a block called nav-link, it’d be BEM’d up like:
```
Comment (rafriki) [inline _posts/2016-05-24-journal-of-a-junior-bem.md:30]: I think nav-link would be an element here, and nav-bar would be the block.

File: _posts/2016-05-24-journal-of-a-junior-bem.md [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+
+<pre><code class="scss">
+.nav-link {} /* Block */
+.nav-link__field {}
+.nav-bar--active {}
```
Comment (rafriki) [inline _posts/2016-05-24-journal-of-a-junior-bem.md]: .nav-bar__nav-link--active perhaps?

File: _posts/2016-05-24-journal-of-a-junior-bem.md (hunk 5 above)
Comment (lukefrake) [inline _posts/2016-05-24-journal-of-a-junior-bem.md]: remove the ',' after block

File: _posts/2016-05-24-journal-of-a-junior-bem.md [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+<pre><code class="scss">
+.block{}
+.block__element{}
+.block--modifier{}
+
+</code></pre>
+<br>
+A block represent an object on your website, and could be a form, header, or navbar for example.
+A block element represents a descendant of your block and is indicated by beginning with a double underscore ‘__ ‘. Elements are what make up your block, and so can be buttons, textboxes or media items.
+A  modifier is a different version of your element or block and is indicated by double hyphen ‘--’. It could be something like --blue that would give a style like background-color: blue. It’ll indicate that the object would be blue (yes, really!). It can also identify states like booleans, or values like bold, normal or thin for properties like font weight.
```
Comment (lukefrake) [inline _posts/2016-05-24-journal-of-a-junior-bem.md]: double space after A

Comment (lukefrake) [inline _posts/2016-05-24-journal-of-a-junior-bem.md]: I wouldn't use `--blue` as an example, blocks elements and modifiers should be away from state where possible. Maybe change it to `--active`

File: _posts/2016-05-24-journal-of-a-junior-bem.md [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+.nav-link {} /* Block */
+.nav-link__field {}
+.nav-bar--active {}
+
+</code></pre>
+<br/>
+
+###What are the benefits of using BEM compared to my previous approach?
+
+My previous approach to stylesheets was very..I’m going to go with amature. I enjoyed writing them but I had no structure to follow. Everything was thrown into one stylesheet and there was a lot of specifics and nesting which I have now learned isn’t a fantastic idea. If I was to change an ID, then I could potentially ruin the style, if I was to change the class of a parent element I could throw off the whole section. There was tons of nesting and so a lot of repeated styles. This lead to a very confusing and daunting stylesheet (but I was proud of it anyways because “Look at all this cool stuff I wrote!”).
```
Comment (lukefrake) [inline _posts/2016-05-24-journal-of-a-junior-bem.md]: 3 .'s for a hyphen. Also maybe amature​ might be amateurish or amateur

Comment (lukefrake) [inline _posts/2016-05-24-journal-of-a-junior-bem.md]: 👍 love this paragraph

**Comment by rafriki** [comment]
great job

File: _posts/2016-05-24-journal-of-a-junior-bem.md (hunk 1 above)
Comment (lukefrake) [inline _posts/2016-05-24-journal-of-a-junior-bem.md:22]: Ignore my point, you explain further on :)

File: _posts/2016-05-24-journal-of-a-junior-bem.md [hunk 10]
```diff
@@ -0,0 +1,79 @@
...
+.right-ear{}
+
+</code></pre>
+<br/>
+
+Reading that and knowing it’s a cat makes sense, but what if you didn’t know? What would the .ear be for - could be a mouse? You can get .fat anything, could it be a fat person? By using BEM you are giving whoever is looking at your markup a good idea of what’s going on and all the different elements relate together but aren't depending on each other!
+
+###And so I conclude..
+
+I really enjoyed learning about BEM (I have this video by Kevin Mack to thank for that), and I can now see why Space Between practice it. It’s informative and I’m really looking forward to applying it to future projects as I can see how much it’s going to benefit me  - it will make me think a lot more about what i’m going to name my classes and help me interpret designs better if i’m thinking of it as blocks and elements. The BEM classnames will help make the markup a lot more descriptive and meaningful especially for my coworkers as we’ll all be following the same syntax.
```
Comment (lukefrake) [inline _posts/2016-05-24-journal-of-a-junior-bem.md]: Can you link to the video here, please? Also, link to his Twitter page or something :)

Comment (lukefrake) [inline _posts/2016-05-24-journal-of-a-junior-bem.md]: i'm should be I'm

File: _posts/2016-05-24-journal-of-a-junior-bem.md (hunk 6 above)
Comment (maggyb) [inline _posts/2016-05-24-journal-of-a-junior-bem.md:30]: I think that's a good point thank you!

File: _posts/2016-05-24-journal-of-a-junior-bem.md [hunk 11]
```diff
@@ -0,0 +1,79 @@
...
+
+The beautiful thing with BEM is that it eliminates a lot of those problems - even though it’s kinda ugly on the outside. It doesn’t need nesting, because a BEM CSS class is independent and doesn’t need tags or IDs to show who it is! It depends on its unique class and its modifiers. 
+
+Another nice quality about BEM is that it’s very informative, you can tell what the style is just by looking at the class. Let’s go with the analogy of a cat:
+
+<pre><code class="scss">
+.cat {}
+.cat__ear{}
+.cat--fat{}
+.cat--fat__ear{}
```
Comment (f000) [inline _posts/2016-05-24-journal-of-a-junior-bem.md:52]: this is wrong, it's called BEM, not BME... see the specs https://en.bem.info/methodology/naming-convention/

Comment (lukefrake) [inline _posts/2016-05-24-journal-of-a-junior-bem.md:52]: I think it depends on where you read, although I do massively prefer the single hyphen / underscore

Alternative ides:
http://getbem.com/naming/
http://csswizardry.com/2013/01/mindbemding-getting-your-head-round-bem-syntax/

Comment (maggyb) [inline _posts/2016-05-24-journal-of-a-junior-bem.md:52]: I don't think it's wrong, it's just an alternative way of doing it. This is the page I did some of my research from
http://csswizardry.com/2013/01/mindbemding-getting-your-head-round-bem-syntax/
if it's really that wrong i'll change it though. This is my first exposure to BEM so i'm not an expert I only know what I've researched

Comment (f000) [inline _posts/2016-05-24-journal-of-a-junior-bem.md:52]: @lukefrake it's not about single hyphen / underscore, I'm ok with our current style

@maggyb ok np, it looks weird

**Comment by lukefrake** [comment]
This is great so I'm deploying 👍 💃 👯 :shipit: 🚀 🎸

**Pull request merged**